src/lib/FieldType/Mapper/FieldTypeToContentTypeStrategy.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\FieldTypeMatrix\FieldType\Mapper;
+
+use Ibexa\Contracts\Core\Repository\ContentTypeService;
+use Ibexa\Contracts\Core\Repository\Values\ContentType\ContentType;
+use Ibexa\Contracts\Core\Repository\Values\ContentType\FieldDefinition;
+
+final class FieldTypeToContentTypeStrategy implements FieldTypeToContentTypeStrategyInterface
+{
+    private ContentTypeService $contentTypeService;
+
+    public function __construct(ContentTypeService $contentTypeService)
+    {
+        $this->contentTypeService = $contentTypeService;
+    }
+
+    public function findContentTypeOf(FieldDefinition $fieldDefinition): ?ContentType
+    {
+        foreach ($this->contentTypeService->loadContentTypeGroups() as $group) {
+            foreach ($this->contentTypeService->loadContentTypes($group) as $type) {
+                $foundFieldDefinition = $type->getFieldDefinition($fieldDefinition->identifier);
+                if ($foundFieldDefinition === null) {
+                    continue;
+                }
+                if ($foundFieldDefinition->id === $fieldDefinition->id) {
+                    return $type;
+                }
+            }
+        }
```

Comment:
Correct me if I'm wrong, but doesn't this code load **all content type groups** and **all content types** in the worst case scenario?

I understand it is how it was before, but it raises a flag for me nonetheless.

Reply:
This was handled like this most likely because `FieldDefinition`  has no information on Content Type.

Reply:
@barw4 Can you possibly check how much effort would be needed to provide Content Type information to `FieldDefinition`? It could be accessed using lazy loaded getter.

Reply:
@webhdx sure. We would have to 
- update `buildFieldDefinitionDomainObject()` https://github.com/ezsystems/ezplatform-kernel/blob/1.3/eZ/Publish/Core/Repository/Mapper/ContentTypeDomainMapper.php#L219 - that includes updating parent calls, therefore tests.
- update `FieldDefinition` VO
- implementing proxy itself in `ContentTypeDomainMapper`.

All those things should be obviously made in `kernel`/`core`.